src/compiler/checker.ts

```diff
@@ -5666,7 +5655,24 @@ namespace ts {
                     }
                 }
 
-                return result & isRelatedTo(sourceReturnType, targetReturnType, reportErrors);
+                result = result & isRelatedTo(sourceReturnType, targetReturnType, reportErrors);

```

Comment:
I think it would be a better idea to first check if they are both predicate types, since it doesn't look like you need to relate the return types in that case.